int-ball2_data_replay/CMakeLists.txt

```
@@ -34,4 +34,6 @@ if(BUILD_TESTING)
   ament_lint_auto_find_test_dependencies()
 endif()
 
-ament_auto_package()
+ament_auto_package(
+  USE_SCOPED_HEADER_INSTALL_DIR
+)
```

I think we should not use this option as generates some path issues apparently.
https://github.com/PickNikRobotics/generate_parameter_library/issues/273

```suggestion
ament_auto_package()
```

---

I got this warning when I build the pkg without `USE_SCOPED_HEADER_INSTALL_DIR`

```
--- stderr: ib2_data_replay                                
In this package, headers install destination is set to `include` by ament_auto_package. It is recommended to install `include/ib2_data_replay` instead and will be the default behavior of ament_auto_package from ROS 2 Kilted Kaiju. On distributions before Kilted, ament_auto_package behaves the same way when you use USE_SCOPED_HEADER_INSTALL_DIR option.
---
```

So that's why I added it. 

I've looked into the github issue that you sent. It seems like it is the `generate_parameter_library` package itself had CMake logic that did not correctly handle the nested include path. So maybe It was making an incorrect assumption about where its own headers would be found.

From my understanding, in our case, when we import header in your main code like `#include "int-ball2_data_replay/bag_pose_replay.hpp"` , `int-ball2_data_replay` is the scoped header directory. If we don't use ament_auto_package(USE_SCOPED_HEADER_iNSTALL_DIR), then when you do colcon build, it probably doesn't know where the bag_pose_replay.hpp is, because ros2 humble will install that header file in `install/<pkg_name>/include` not `install/pkg_name/include/<pkg_name>`.

This is reference that I found -> https://zenn.dev/tiryoh/articles/2025-05-23-ament-cmake-auto

Let me know if I misunstood something.

---

I see. That warning error might be because we are using the `ament_auto_add_library` function.
I have never seen this before, when using the `add_library` manually.
This auto method might save some lines but when porting to Jazzy, it make cause some trouble. Don't you think?
`USE_SCOPED_HEADER_INSTALL_DIR` seems to be a new function an it should take some time to the fixes to be applied. 
https://github.com/ament/ament_cmake/pull/596

Maybe we should wait a little bit til such transition, and keep the `add_library` function.

---

I think it's and already implemented for ros2 humble (https://github.com/ament/ament_cmake/pull/577). The warning [PR](https://github.com/ament/ament_cmake/pull/596) that you sent was intentionally added in Humble to people to know the change that will be happened in the newer distro. So once we migrate our environment to Jazzy or more modern version, we can just remove that flag, since the header installed path will be updated as a default.

So whether we go back to use `add_library` or just remove the flag `USE_SCOPED_HEADER_INSTALL_DIR` in the new distro, we can work on it in the new branch after we finish merging this branch.

---

Thank you for the clarification.
In that case, we can just keep the `USE_SCOPED_HEADER_INSTALL_DIR` implementation until any bug is found.
We might just need to keep a look that there is no problem with that.